backend/curriculum_tracking/tests/test_AgileCard.py

```diff
@@ -788,4 +788,26 @@ def test_request_review_but_no_review_done_since_time_of_review_request(self):
         # so I manually created a 'recruit_project' attribute.
         project_two.recruit_project = project_two
 
-        self.assertEqual(AgileCard.get_users_that_reviewed_since_last_review_request(project_two), [])
\ No newline at end of file
+        self.assertEqual(AgileCard.get_users_that_reviewed_since_last_review_request(project_two), [])
+
+
+class RepoUrlTests(TestCase):
+
+    def setUp(self):
+        self.card = factories.AgileCardFactory()
```

Comment:
setup should be used for common setup. MEaning, for setting up things that are needed for multiple tests. 

you have defined tewo cards and two tests.  card 1 is only used in test 1, card 2 is only used in test 2.  So card 1 should be defined inside test 1 and card 2 should be defined inside test 2.

This is because cohesive code is good.

Also it makes your code much less leaky and makes the intentions clear.  If you put card 1 in setup then you are sayin "I intend to use this card in multiple tests". But that is not your intention.  Why should test 2 have access to card 1 if it actually doersn't need it? (principle of least privilage)